src/components/AppHeader.tsx

```diff
@@ -1,16 +1,27 @@
+import { useState } from "react"
+import { useIsMobile } from "../assets/customHooks/useIsMobile"
 import { data } from "../data/data"
+import HamburgerMenu from "./HamburgerMenu"
 
 const AppHeader = () => {
+    const isHamburgerToShow = useIsMobile(600)
+    const [isShowHamburger, setIsShowHamburger] = useState(false)
 
+    const toggleHamburger = () => {
+        setIsShowHamburger(prevIsShow => !prevIsShow);
+    }
 
     return (
         <div className="header-container flex align-center">
-            <img className="hamburger" src={data.header.hamburger.src} alt="hamburger-menu" />
+            <button onClick={toggleHamburger} className="open-burger">
+                <img className="hamburger" src={data.header.hamburger.src} alt="hamburger-menu" />
+            </button>
+            {isShowHamburger && isHamburgerToShow && <HamburgerMenu setIsShowHamburger={setIsShowHamburger} />}
```

Comment:
not so clear just form reading this, why two isShowHamburger in different syntax